Sources/ClientRuntime/Networking/Streaming/Stream.swift

```diff
@@ -61,48 +65,10 @@ public enum StreamError: Error {
     case notSupported(String)
 }
 
-extension Stream {
-    func isEqual(to other: Stream) throws -> Bool {
-        let selfData = try readToEnd()
-        let otherData = try other.readToEnd()
-        return selfData == otherData
-    }
-}
-
 extension Stream {
     public func seek(toOffset offset: Int) throws {
         guard isSeekable else {
             throw StreamError.notSupported("Seeking is not supported.")
         }
     }
 }
-
-extension Stream {
```

Comment:
This was the not-really-asynchronous / polling stream read method that was causing CPU problems.

Instead of providing this method in a protocol extension, it's now implemented individually in the two stream types.